Formula/ox.rb

```diff
@@ -25,17 +25,6 @@ def install
   end
 
   test do
-    require "pty"
-
-    _, w, pid = PTY.spawn(bin/"ox", "test.txt")
-    sleep 1
-    w.write "Hello Homebrew!\n"
-    w.write "\cS"
-    sleep 1
-    w.write "\cQ"
-
-    assert_match "Hello Homebrew!\n", (testpath/"test.txt").read
```

Comment:
this test runs out fine with 0.2.5, but keep getting error with 0.2.7 as error shown as below:

```
  Errno::ENOENT: No such file or directory @ rb_sysopen - /private/tmp/ox-test-20230703-23774-1kutdr4/test.txt
```

Reply:
That's Ruby failing to read the test file, presumably because it's not been created properly. What's the full output of `brew test --verbose --debug ox` locally?

I'd hold off merging until this is verified.

Reply:
```
==> Testing ox
/opt/homebrew/Library/Homebrew/test.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/ox.rb
==> Temporary files retained at:
/private/tmp/ox-test-20230705-94450-r4j4br
Error: ox: failed
An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ rb_sysopen - /private/tmp/ox-test-20230705-94450-r4j4br/test.txt
```

Reply:
but if i ran manually, I can do the ctrl+s and ctrl+q successfully in my local

Reply:
Would be good to figure out what, if anything, is being sent by `ox` to `stdout` or `stderr`.

Reply:
Any guide to debug thru this?

Reply:
```ruby
# add `read` so we can see stdout/stderr
read, write, pid = PTY.spawn(bin/"ox", "test.txt")

# existing code here

# output stdout/stderr
pp read
```

Reply:
`pp read` will return `#<File:/dev/ttys002>`. Not sure what to make out of that information.

I played around with this but was not able to fix it :(

Reply:
Can get test to pass if I change `TERM` and force a read on macOS, e.g.
```diff
-    _, w, pid = PTY.spawn(bin/"ox", "test.txt")
+    ENV["TERM"] = "xterm"
+    r, w, pid = PTY.spawn(bin/"ox", "test.txt")
     sleep 1
     w.write "Hello Homebrew!\n"
     w.write "\cS"
     sleep 1
     w.write "\cQ"
+    r.read
```

`TERM` is because `term` crate doesn't define colors for `dumb` terminal while values like `xterm` default to ANSI. `ox` fails trying to unwrap None at https://github.com/curlpipe/ox/blob/0.2.7/src/terminal.rs#L82-L83
```rust
        } else if let Ok(info) = TermInfo::from_env() {
            if info.numbers.get("colors").unwrap() == &256 {
```

Haven't checked why it needs to read. It fails if I remove it.

Reply:
I am gonna adopt @cho-m's patch and get this PR going.